### PR TITLE
Save localStorage and sessionStorage data from browser

### PR DESCRIPTION
**Pull request by alzamer2**
this update will make  _restore_cookies and _apply_cookies save localstorage and sessionStorage from the Browser as will as some login information (like Bearer token is being saved there and this will make it easier to synchronize  those informations

### CODE REVIEW

**Comment by alzamer2** [comment]
at last 
man there a lot of lint test

**Comment by dipu-bd** [comment]
Sources are downloaded automatically by the app whenever a new version is available in the repository. This auto download feature will not work, if wuxiacom.py depends on wuxiacom.proto as well. This is the reason why we had to put the proto string at the end of the python file.

File: sources/en/w/wuxiacom.py [hunk 1]
```diff
@@ -351,8 +351,4 @@ def __repr__(self):
...
-"""  # noqa: E501
-)
+WUXIWORLD_PROTO = json.loads(open('wuxiacom.proto').read())  # noqa: E501
```
Comment (dipu-bd) [inline sources/en/w/wuxiacom.py:354]: bring it back again please, we need it in one single file to make it work.

File: lncrawl/core/browser.py [hunk 2]
```diff
@@ -72,36 +75,52 @@ def _init_browser(self):
...
+                "var ls = window.localStorage, items = {}; "
+                "for (var i = 0, k; i < ls.length; ++i) "
+                "  items[k = ls.key(i)] = ls.getItem(k); "
+                "return items; ")
+            self.browser_storage['sessionStorage'] = self._driver.execute_script(
+                "var ls = window.sessionStorage, items = {}; "
+                "for (var i = 0, k; i < ls.length; ++i) "
+                "  items[k = ls.key(i)] = ls.getItem(k); "
+                "return items; ")
+            logger.debug("Storage retrieved: %s", self.browser_storage)
```
Comment (dipu-bd) [inline lncrawl/core/browser.py:124]: good idea to store sessionStorage and localStorage along with the cookies. thanks

**Comment by alzamer2** [comment]
done

**Comment by dipu-bd** [comment]
150
to ignore linting errors, you just have to add `# noqa` at the end of the line. it was already there for wuxia proto string

**Comment by alzamer2** [comment]
> 150 to ignore linting errors, you just have to add `# noqa` at the end of the line. it was already there for wuxia proto string

ohh i did not know that thanks for info

**Comment by dipu-bd** [comment]
Merged manually to `dev` branch

**Comment by alzamer2** [comment]
> Merged manually to `dev` branch

Do you want me to do a pull request to dev branch?

**Pull request closed without merging**